### PR TITLE
Frontend changes

### PR DESCRIPTION
**Pull request by UAMS16**
Features table didnt have message for no results, dna feature profile table location link now identifiable. These are a few of the changes made in this PR. We have made many changes the past few months so we are applying those changes to other pages as well
image
image
image
image

### CODE REVIEW

File: cegs_portal/search/templates/search/v1/experiment.html [hunk 1]
```diff
@@ -193,8 +193,9 @@
                 <div class="facet-content-section container">
                     <form name="regionUploadForm">
                         <div>
-                            <label class="font-bold" for="regionFile">Highlight Regions</label>
-                            <input id="regionUploadInput" type="file" accept=".bed" name="regionFile" />
+                            <label class="font-bold flex justify-center" for="regionFile">Highlight Regions</label>
+                            <input class="hidden" id="regionUploadInput" type="file" accept=".bed" name="regionFile"/>
+                            <label for="regionUploadInput" class="flex justify-center search-button mt-3 border-2 p-2 text-sm">Upload File</label>
```
Comment (tncowart) [inline cegs_portal/search/templates/search/v1/experiment.html:198]: This is a surprising pattern

File: cegs_portal/search/templates/search/v1/experiment.html [hunk 2]
```diff
@@ -231,7 +234,7 @@
                     </div>
                     <div>
                         <label for="covSelect">Choose a Coverage View:</label>
-                        <select name="views" id="covSelect">
+                        <select class="search-button border-2 p-2 h-10 text-sm" name="views" id="covSelect">
```
Comment (tncowart) [inline cegs_portal/search/templates/search/v1/experiment.html:237]: Given how many places it's used that aren't searching, I think it's time to rename the `search-button` class.

File: cegs_portal/search/templates/search/v1/partials/_features.html [hunk 3]
```diff
@@ -2,70 +2,74 @@
...
-                    <!--! Font Awesome Pro 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2023 Fonticons, Inc. -->
-                    <path
-                        d="M416 0c17.7 0 32 14.3 32 32c0 59.8-30.3 107.5-69.4 146.6c-28 28-62.5 53.5-97.3 77.4l-2.5 1.7c-11.9 8.1-23.8 16.1-35.5 23.9l0 0 0 0 0 0-1.6 1c-6 4-11.9 7.9-17.8 11.9c-20.9 14-40.8 27.7-59.3 41.5H283.3c-9.8-7.4-20.1-14.7-30.7-22.1l7-4.7 3-2c15.1-10.1 30.9-20.6 46.7-31.6c25 18.1 48.9 37.3 69.4 57.7C417.7 372.5 448 420.2 448 480c0 17.7-14.3 32-32 32s-32-14.3-32-32H64c0 17.7-14.3 32-32 32s-32-14.3-32-32c0-59.8 30.3-107.5 69.4-146.6c28-28 62.5-53.5 97.3-77.4c-34.8-23.9-69.3-49.3-97.3-77.4C30.3 139.5 0 91.8 0 32C0 14.3 14.3 0 32 0S64 14.3 64 32H384c0-17.7 14.3-32 32-32zM338.6 384H109.4c-10.1 10.6-18.6 21.3-25.5 32H364.1c-6.8-10.7-15.3-21.4-25.5-32zM109.4 128H338.6c10.1-10.7 18.6-21.3 25.5-32H83.9c6.8 10.7 15.3 21.3 25.5 32zm55.4 48c18.4 13.8 38.4 27.5 59.3 41.5c20.9-14 40.8-27.7 59.3-41.5H164.7z"
-                    />
-                </svg>
-                DNA Features
+        {% if features|length > 0 %}
```
Comment (tncowart) [inline cegs_portal/search/templates/search/v1/partials/_features.html:5]: Shouldn't this be above the container div (line 4)?

File: cegs_portal/search/templates/search/v1/reg_effect.html [hunk 4]
```diff
@@ -80,16 +80,15 @@ <h1>
 
     {% if regulatory_effect.sources.exists %}
     <div class="container">
-        <div class="flex max-w-full min-w-full justify-between items-center">
-            <div class="justify-self-start"></div>
-            <div class="table-title flex items-center justify-center">
+        <div class="table-title" style="display: flex; align-items: center; justify-content: space-between; position: relative;">
+            <div style="display: flex; align-items: center; justify-content: center; flex-grow: 1;">
```
Comment (tncowart) [inline cegs_portal/search/templates/search/v1/reg_effect.html:84]: Please don't use inline style tags. Either use the relevant tailwind classes or, if you find you're using the same group of styles over and over, put those styles in a new CSS class and use that class.

File: cegs_portal/search/templates/search/v1/reg_effect.html [hunk 5]
```diff
@@ -106,9 +105,8 @@ <h1>
 
     {% if regulatory_effect.targets.exists %}
     <div class="container">
-        <div class="flex max-w-full min-w-full justify-between items-center">
-            <div class="justify-self-start"></div>
-            <div class="table-title flex items-center justify-center">
+        <div class="table-title" style="display: flex; align-items: center; justify-content: space-between; position: relative;">
+            <div style="display: flex; align-items: center; justify-content: center; flex-grow: 1;">
```
Comment (tncowart) [inline cegs_portal/search/templates/search/v1/reg_effect.html:109]: Please don't use inline style tags. Either use the relevant tailwind classes or, if you find you're using the same group of styles over and over, put those styles in a new CSS class and use that class.

File: cegs_portal/search/templates/search/v1/partials/_features.html [hunk 6]
```diff
@@ -2,70 +2,74 @@
...
-                    <a href="?page={{ features.next_page_number }}&dist={{ dist }}{% for type in feature_types %}&feature_type={{ type }}{% endfor %}&assembly={{ assembly|default:'GRCh38' }}" hx-get="{% url 'search:dna_feature_loc' loc.chr loc.start loc.end %}?page={{ features.next_page_number }}&dist={{ dist }}{% for type in feature_types %}&feature_type={{ type }}{% endfor %}&assembly={{ assembly|default:'GRCh38' }}" hx-target="#features-table">next</a>
-                    <a href="?page={{ features.paginator.num_pages }}&dist={{ dist }}{% for type in feature_types %}&feature_type={{ type }}{% endfor %}&assembly={{ assembly|default:'GRCh38' }}" hx-get="{% url 'search:dna_feature_loc' loc.chr loc.start loc.end %}?page={{ features.paginator.num_pages }}&dist={{ dist }}{% for type in feature_types %}&feature_type={{ type }}{% endfor %}&assembly={{ assembly|default:'GRCh38' }}" hx-target="#features-table">last »</a>
-                {% endif %}
-            </span>
-        </div>
-
+            </div>
     </div>
+        {% else %}
+            <div class="table-title" style="display: flex; align-items: center; justify-content: center">
```
Comment (tncowart) [inline cegs_portal/search/templates/search/v1/partials/_features.html:72]: Please don't use inline style tags. Either use the relevant tailwind classes or, if you find you're using the same group of styles over and over, put those styles in a new CSS class and use that class.

**Review by tncowart — CHANGES_REQUESTED** [review]
Visually, these changes look really good! There are some code things I'd like changed though.

File: cegs_portal/search/templates/search/v1/dna_feature.html [hunk 7]
```diff
@@ -63,10 +62,9 @@ <h1>
                 <td>
                     <a href="{% url 'search:results' %}?query={{ feature.chrom_name }}:{{ feature.location.lower }}-{{ feature.location.upper|add:"-1" }}">
                         <span>{{ feature.chrom_name }}</span>:
-                        <span>{{ feature.location.lower|intcomma }} - {{ feature.location.upper|add:"-1"|intcomma }}</span>
+                        <span>{{ feature.location.lower|intcomma }} - {{ feature.location.upper|add:"-1"|intcomma }}{% if feature.strand %}:{{ feature.strand }}{% endif %}</span>
```
[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/templates/search/v1/dna_feature.html:65]: We are using the if_strand helper in other places. I can remove the if statement and use the helper

File: cegs_portal/search/templates/search/v1/reg_effect.html (hunk 4 above)
Comment (UAMS16) [reply cegs_portal/search/templates/search/v1/reg_effect.html:84]: RIght, just carried over some inline css. Seems like there are places that havent been swapped with tailwind. Will look further into this

File: cegs_portal/search/templates/search/v1/reg_effect.html [hunk 8]
```diff
@@ -80,16 +80,15 @@ <h1>
...
+        <div class="table-title table-container-header">
+            <div class="table-container-title">
                 <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" class="svg-icon mr-1.5">
                     <!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2023 Fonticons, Inc. -->
                     <path d="M315.4 15.5C309.7 5.9 299.2 0 288 0s-21.7 5.9-27.4 15.5l-96 160c-5.9 9.9-6.1 22.2-.4 32.2s16.3 16.2 27.8 16.2H384c11.5 0 22.2-6.2 27.8-16.2s5.5-22.3-.4-32.2l-96-160zM288 312V456c0 22.1 17.9 40 40 40H472c22.1 0 40-17.9 40-40V312c0-22.1-17.9-40-40-40H328c-22.1 0-40 17.9-40 40zM128 512a128 128 0 1 0 0-256 128 128 0 1 0 0 256z"/>
                 </svg>
                 Tested Elements
             </div>
-            <div class="justify-self-end">
+            <div style="position: absolute; right: 0;">
```
Comment (tncowart) [inline cegs_portal/search/templates/search/v1/reg_effect.html:91]: Maybe this should also be a class, like `table-container-downloads`.

**Review by tncowart — CHANGES_REQUESTED** [review]
Almost there!

File: cegs_portal/search/templates/search/v1/dna_feature.html [hunk 9]
```diff
@@ -259,29 +254,29 @@ <h1>
...
-                        <th class="chrom-first-end-cap chrom-light-band">Accession ID</th>
-                        <th class="chrom-dark-band">Type</th>
-                        <th class="chrom-light-band">Cell Line</th>
-                        <th class="chrom-light-band chrom-right-centromere">Location</th>
-                        <th class="chrom-light-band chrom-left-centromere">Strand</th>
-                        <th class="chrom-last-end-cap chrom-dark-band">Reference Genome</th>
+                        <th class="chrom-first-end-cap chrom-dark-band">Type</th>
+                        <th class="chrom-light-band chrom-right-centromere  ">Cell Line</th>
+                        <th class="chrom-left-centromere chrom-light-band">Location</th>
+                        <th class="chrom-last-end-cap chrom-light-band">Feature Details</th>
```
Comment (tncowart) [inline cegs_portal/search/templates/search/v1/dna_feature.html:260]: I think just `Details`? We know these are DNA Features from the table container header. That also matches the place we already do this.

**Review by tncowart — APPROVED** [review]

**Pull request merged**